### PR TITLE
chore: add tests to check license propagates to console

### PR DESCRIPTION
**Pull request by alejandroEsc**
Adding test here, checking to see if we can show failure today and then we can test against branch with fix.

### CODE REVIEW

**Comment by alejandroEsc** [comment]
The tests pass because we are passing a values file that does not have the accurate values. Ill update this, none of these ci values ran `-test-license-with-console` which shoud have failed.

File: .github/workflows/pull_requests_redpanda.yaml [hunk 1]
```diff
@@ -149,6 +156,13 @@ jobs:
         run: |
           .github/create-sasl-secret.sh "some-users"
 
+      - name: Create license secret
+        if: steps.list-changed.outputs.changed == 'true'
+        env:
+          REDPANDA_LICENSE: ${{ secrets.REDPANDA_SAMPLE_LICENSE }}
+        run: |
+          .github/create-license-secret.sh "redpanda-license" "%REDPANDA_LICENSE%"
```
Comment (RafalKorepta) [inline .github/workflows/pull_requests_redpanda.yaml:164]: I think this doesn't work as tests shows:

```
2023-09-19T19:20:50.2924650Z >>> kubectl --request-timeout=30s logs redpanda-439416s7xb-configuration-7lkkd --namespace redpanda-439416s7xb --container redpanda-post-install
2023-09-19T19:20:50.4001383Z unable to set license: request PUT https://redpanda-439416s7xb-0.redpanda-439416s7xb.redpanda-439416s7xb.svc.cluster.local.:9644/v1/features/license failed: Bad Request, body: "{\"message\": \"License is malformed: Outer envelope malformed\", \"code\": 400}"
```

https://github.com/redpanda-data/helm-charts/actions/runs/6239380055/job/16938410422?pr=734#step:20:27440

**Comment by RafalKorepta** [comment]
https://github.com/redpanda-data/helm-charts/actions/runs/6239380061/job/16937188683?pr=734#step:12:6352
```
  + rpk topic create test -v
  18:31:21.636  DEBUG  opening connection to broker  {"addr": "cluster-tls-0.cluster-tls.operator-9t146hytr9.svc.cluster.local:9092", "broker": "seed_0"}
  18:31:21.647  DEBUG  connection opened to broker  {"addr": "cluster-tls-0.cluster-tls.operator-9t146hytr9.svc.cluster.local:9092", "broker": "seed_0"}
  18:31:21.647  DEBUG  issuing api versions request  {"broker": "seed_0", "version": 3}
  18:31:21.647  DEBUG  wrote ApiVersions v3  {"broker": "seed_0", "bytes_written": 31, "write_wait": "33.1µs", "time_to_write": "60.4µs", "err": null}
  18:31:21.648  DEBUG  read ApiVersions v3  {"broker": "seed_0", "bytes_read": 296, "read_wait": "72.599µs", "time_to_read": "429.5µs", "err": null}
  18:31:21.648  DEBUG  connection initialized successfully  {"addr": "cluster-tls-0.cluster-tls.operator-9t146hytr9.svc.cluster.local:9092", "broker": "seed_0"}
  18:31:21.648  DEBUG  wrote Metadata v7  {"broker": "seed_0", "bytes_written": 22, "write_wait": "12.969286ms", "time_to_write": "98.2µs", "err": null}
  18:31:21.656  DEBUG  read Metadata v7  {"broker": "seed_0", "bytes_read": 299, "read_wait": "92.1µs", "time_to_read": "7.736791ms", "err": null}
  18:31:21.656  DEBUG  opening connection to broker  {"addr": "cluster-tls-0.cluster-tls.operator-9t146hytr9.svc.cluster.local.:9092", "broker": "0"}
  18:31:21.658  WARN  unable to open connection to broker  {"addr": "cluster-tls-0.cluster-tls.operator-9t146hytr9.svc.cluster.local.:9092", "broker": "0", "err": "dial tcp: lookup cluster-tls-0.cluster-tls.operator-9t146hytr9.svc.cluster.local. on 10.96.0.10:53: no such host"}
  unable to create topics [test]: unable to dial: dial tcp: lookup cluster-tls-0.cluster-tls.operator-9t146hytr9.svc.cluster.local. on 10.96.0.10:53: no such host
```

I will try to mitigate this problem.

File: .github/workflows/pull_requests_redpanda.yaml (hunk 1 above)
Comment (alejandroEsc) [reply .github/workflows/pull_requests_redpanda.yaml:164]: Yes I noticed this, was about to start to track down the issue, thanks for looking into it.

**Comment by alejandroEsc** [comment]
OK correct tests are running/passing and secret is being pull in using the new license items. The tests ping console, so we know they are reaching console.

**Review by RafalKorepta — CHANGES_REQUESTED** [review]
```
2023-09-20T19:03:25.7231757Z testing that we do NOT have an open source license
2023-09-20T19:03:25.7233605Z ++ jq -r .console.license.type
2023-09-20T19:03:25.7234339Z ++ curl http://redpanda-console:8080/api/cluster/overview
2023-09-20T19:03:25.7235182Z   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
2023-09-20T19:03:25.7235782Z                                  Dload  Upload   Total   Spent    Left  Speed
2023-09-20T19:03:25.7236152Z 
2023-09-20T19:03:25.7236540Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2023-09-20T19:03:25.7237200Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
2023-09-20T19:03:25.7237727Z   0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0
2023-09-20T19:03:25.7238274Z   0     0    0     0    0     0      0      0 --:--:--  0:00:03 --:--:--     0
2023-09-20T19:03:25.7239309Z   0     0    0     0    0     0      0      0 --:--:--  0:00:04 --:--:--     0curl: (6) Could not resolve host: redpanda-console
2023-09-20T19:03:25.7240186Z license test passed.
2023-09-20T19:03:25.7240601Z + type=
2023-09-20T19:03:25.7241027Z + '[' '' == open_source ']'
2023-09-20T19:03:25.7241400Z + set +x
```

https://github.com/redpanda-data/helm-charts/actions/runs/6252827340/job/16976940583?pr=734#step:20:19659

**Comment by alejandroEsc** [comment]
> ```
> 2023-09-20T19:03:25.7231757Z testing that we do NOT have an open source license
> 2023-09-20T19:03:25.7233605Z ++ jq -r .console.license.type
> 2023-09-20T19:03:25.7234339Z ++ curl http://redpanda-console:8080/api/cluster/overview
> 2023-09-20T19:03:25.7235182Z   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
> 2023-09-20T19:03:25.7235782Z                                  Dload  Upload   Total   Spent    Left  Speed
> 2023-09-20T19:03:25.7236152Z 
> 2023-09-20T19:03:25.7236540Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
> 2023-09-20T19:03:25.7237200Z   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
> 2023-09-20T19:03:25.7237727Z   0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0
> 2023-09-20T19:03:25.7238274Z   0     0    0     0    0     0      0      0 --:--:--  0:00:03 --:--:--     0
> 2023-09-20T19:03:25.7239309Z   0     0    0     0    0     0      0      0 --:--:--  0:00:04 --:--:--     0curl: (6) Could not resolve host: redpanda-console
> 2023-09-20T19:03:25.7240186Z license test passed.
> 2023-09-20T19:03:25.7240601Z + type=
> 2023-09-20T19:03:25.7241027Z + '[' '' == open_source ']'
> 2023-09-20T19:03:25.7241400Z + set +x
> ```
> 
> https://github.com/redpanda-data/helm-charts/actions/runs/6252827340/job/16976940583?pr=734#step:20:19659

that is very strange....

**Review by RafalKorepta — APPROVED** [review]
LGTM

**Pull request merged**